adapters/repos/db/lsmkv/segment_group.go

```diff
@@ -516,6 +516,55 @@ func (sg *SegmentGroup) getWithUpperSegmentBoundary(key []byte, segments []*segm
 	return nil, nil
 }
 
+func (sg *SegmentGroup) exists(key []byte) (bool, error) {
+	beforeMaintenanceLock := time.Now()
+	segments, release := sg.getAndLockSegments()
+	defer release()
+
+	if time.Since(beforeMaintenanceLock) > 100*time.Millisecond {
+		sg.logger.WithField("duration", time.Since(beforeMaintenanceLock)).
+			WithField("action", "lsm_segment_group_get_obtain_maintenance_lock").
+			Debug("waited over 100ms to obtain maintenance lock in segment group get()")
+	}
+
+	return sg.existsWithUpperSegmentBoundary(key, segments)
+}
+
+// not thread-safe on its own, as the assumption is that this is called from a
+// lockholder, e.g. within .exists()
+func (sg *SegmentGroup) existsWithUpperSegmentBoundary(key []byte, segments []*segment) (bool, error) {
+	// assumes "replace" strategy
+
+	// start with latest and exit as soon as something is found, thus making sure
+	// the latest takes presence
```

Comment:
just a typo in the comment `precedence`

Reply:
hmm, good point. I see this typo across the repo ([example](https://github.com/weaviate/weaviate/blob/8d206fa19d1e35a2afa936bf1f7634342748b643/adapters/repos/db/lsmkv/entities/strategies.go#L15)) and now it makes me think if it is really a typo or intentional :D

Reply:
aham, I think it might be better two change in both them?

Reply:
yeah, fixed all the places (6 actually :D )